tests/model/api/test_http_api_generator.py

```diff
@@ -76,6 +77,57 @@ def test_auth_intrinsic_default_auth(self):
         with pytest.raises(InvalidResourceException):
             HttpApiGenerator(**self.kwargs)._construct_http_api()
 
+    def test_auth_iam_enabled(self):
```

Comment:
This goes back to my validation comment. This might not be the right place to assert that happens but we should have some negative test cases (string passed, dict, etc). The main reason is that CFN send json data to a CFN Macro, which is how SAM runs, and CFN doesn't do any validation on that input. So we need to make sure we are guarding our code properly from input way may receive in that json data.

Reply:
Agree, on a general note we need to test against many more bad/wrong inputs

Reply:
Do y'all want me to write more tests? If so, I'd appreciate some guidance on which tests to write.

Reply:
You could add a `test_auth_iam_not_enabled_with_unsupported_values(self)`
It would loop on an array of values and check for each one that no IAM auth is added.
Examples of non boolean values that we could get as input:
`unsupported_values = [1, "", [], {}, { "Ref": "MyVar" }, "True", None]`

Reply:
Got it. Fairly easy change.

Reply:
Turns out this caught an issue so great suggestion!